pkg/controller/external_nofork.go

```diff
@@ -111,15 +114,16 @@ func getJSONMap(mg xpresource.Managed) (map[string]any, error) {
 }
 
 type noForkExternal struct {
-	ts             terraform.Setup
-	resourceSchema *schema.Resource
-	config         *config.Resource
-	instanceDiff   *tf.InstanceDiff
-	params         map[string]any
-	rawConfig      cty.Value
-	logger         logging.Logger
-	metricRecorder *metrics.MetricRecorder
-	opTracker      *AsyncTracker
+	ts                             terraform.Setup
+	resourceSchema                 *schema.Resource
+	config                         *config.Resource
+	instanceDiff                   *tf.InstanceDiff
+	params                         map[string]any
+	initProviderExclusiveParamKeys []string
```

Comment:
I think we don't need to compute this in external connector's `Connect` and thus, we can get rid of the new member here.